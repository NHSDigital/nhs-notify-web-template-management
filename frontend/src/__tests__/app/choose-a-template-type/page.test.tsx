import { render } from '@testing-library/react';
import ChooseATemplateTypePage from '@app/choose-a-template-type/page';
import { TemplateFormState } from 'nhs-notify-web-template-management-utils';
import { useActionState } from 'react';

jest.mock('next/navigation', () => ({
  redirect: () => {
    throw new Error('Simulated redirect');
  },

  RedirectType: {
    push: 'push',
    replace: 'replace',
  },
}));

jest.mock('react', () => {
  const originalModule = jest.requireActual('react');

  return {
    ...originalModule,
    useActionState: (
      _: (
        formState: TemplateFormState,
        formData: FormData
      ) => Promise<TemplateFormState>,
      initialState: TemplateFormState
    ) => [initialState, '/action'],
  };
});

test('ChooseATemplateTypePage', async () => {
  const page = await ChooseATemplateTypePage();

  const container = render(page);

  expect(container.asFragment()).toMatchSnapshot();
});
