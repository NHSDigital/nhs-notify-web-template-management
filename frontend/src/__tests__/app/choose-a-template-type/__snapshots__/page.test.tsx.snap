// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChooseATemplateTypePage 1`] = `
<DocumentFragment>
  <div
    class="nhsuk-back-link"
  >
    <a
      class="nhsuk-back-link__link"
      href="/templates/manage-templates"
    >
      <svg
        aria-hidden="true"
        class="nhsuk-icon nhsuk-icon__chevron-left"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.5 12c0-.3.1-.5.3-.7l5-5c.4-.4 1-.4 1.4 0s.4 1 0 1.4L10.9 12l4.3 4.3c.4.4.4 1 0 1.4s-1 .4-1.4 0l-5-5c-.2-.2-.3-.4-.3-.7z"
        />
      </svg>
      Back to all templates
    </a>
  </div>
  <main
    class="nhsuk-main-wrapper"
    id="maincontent"
    role="main"
  >
    <form
      action="/action"
    >
      <input
        name="form-id"
        readonly=""
        type="hidden"
        value="choose-a-template-type"
      />
      <input
        name="csrf_token"
        readonly=""
        type="hidden"
        value="no_token"
      />
      <div
        class="nhsuk-form-group"
      >
        <fieldset
          class="nhsuk-fieldset"
        >
          <legend
            class="nhsuk-fieldset__legend nhsuk-fieldset__legend--l"
            data-testid="templateType-form__legend"
          >
            <h1
              class="nhsuk-fieldset__heading"
            >
              Choose a template type to create
            </h1>
          </legend>
          <p>
            <a
              href="/features"
              rel="noopener noreferrer"
              target="_blank"
            >
              Learn more about message channels (opens in a new tab)
            </a>
          </p>
          <div
            class="nhsuk-form-group"
          >
            <div
              class="nhsuk-hint"
              id="templateType--hint"
            >
              Select one option
            </div>
            <div
              aria-describedby="templateType--hint"
              class="nhsuk-radios"
              id="templateType"
            >
              <div
                class="nhsuk-radios__item"
              >
                <input
                  class="nhsuk-radios__input"
                  data-testid="NHS_APP-radio"
                  id="templateType-NHS_APP"
                  name="templateType"
                  type="radio"
                  value="NHS_APP"
                />
                <label
                  class="nhsuk-label nhsuk-radios__label"
                  for="templateType-NHS_APP"
                  id="templateType-NHS_APP--label"
                >
                  NHS App message
                </label>
              </div>
              <div
                class="nhsuk-radios__item"
              >
                <input
                  class="nhsuk-radios__input"
                  data-testid="EMAIL-radio"
                  id="templateType-EMAIL"
                  name="templateType"
                  type="radio"
                  value="EMAIL"
                />
                <label
                  class="nhsuk-label nhsuk-radios__label"
                  for="templateType-EMAIL"
                  id="templateType-EMAIL--label"
                >
                  Email
                </label>
              </div>
              <div
                class="nhsuk-radios__item"
              >
                <input
                  class="nhsuk-radios__input"
                  data-testid="SMS-radio"
                  id="templateType-SMS"
                  name="templateType"
                  type="radio"
                  value="SMS"
                />
                <label
                  class="nhsuk-label nhsuk-radios__label"
                  for="templateType-SMS"
                  id="templateType-SMS--label"
                >
                  Text message (SMS)
                </label>
              </div>
            </div>
          </div>
        </fieldset>
      </div>
      <button
        aria-disabled="false"
        class="nhsuk-button"
        data-testid="submit-button"
        id="choose-a-template-type-submit-button"
        type="submit"
      >
        Continue
      </button>
    </form>
  </main>
</DocumentFragment>
`;
